src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/RuleManagerImpl.java

```diff
@@ -0,0 +1,146 @@
+/* Copyright 2021 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+import static org.openkilda.Constants.TableNumber.INGRESS_TABLE_ID;
+import static org.openkilda.Constants.TableNumber.INPUT_TABLE_ID;
+import static org.openkilda.Constants.TableNumber.POST_INGRESS_TABLE_ID;
+import static org.openkilda.Constants.TableNumber.TRANSIT_TABLE_ID;
+import static org.openkilda.model.cookie.Cookie.DROP_RULE_COOKIE;
+import static org.openkilda.model.cookie.Cookie.MULTITABLE_INGRESS_DROP_COOKIE;
+import static org.openkilda.model.cookie.Cookie.MULTITABLE_POST_INGRESS_DROP_COOKIE;
+import static org.openkilda.model.cookie.Cookie.MULTITABLE_TRANSIT_DROP_COOKIE;
+
+import org.openkilda.factory.DefaultRulesGeneratorFactory;
+import org.openkilda.factory.FlowRulesGeneratorFactory;
+import org.openkilda.factory.RuleGenerator;
+import org.openkilda.model.FlowPath;
+import org.openkilda.model.Switch;
+import org.openkilda.model.SwitchId;
+import org.openkilda.model.SwitchProperties;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class RuleManagerImpl implements RuleManager {
+
+    DefaultRulesGeneratorFactory defaultRulesFactory;
+    FlowRulesGeneratorFactory flowRulesFactory;
+
+    public RuleManagerImpl(RuleManagerConfig config) {
+        defaultRulesFactory = new DefaultRulesGeneratorFactory(config);
+        flowRulesFactory = new FlowRulesGeneratorFactory();
+    }
+
+    @Override
+    public List<Command> buildRulesForFlowPath(DataAdapter adapter) {
+        // todo: implement build rules for path
+        return null;
+    }
+
+    @Override
+    public List<Command> buildRulesForSwitch(DataAdapter adapter) {
+        validateAdapter(adapter);
+
+        Switch sw = adapter.getSwitches().values().stream().findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("No switches found"));
```

Comment:
this looks a bit misleading, we are looking for one switch, but Exception refers to many

Reply:
Reworked.